daemon/containerd/image_list.go

```diff
@@ -291,26 +300,48 @@ func (i *ImageService) setupFilters(ctx context.Context, imageFilters filters.Ar
 		return err
 	})
 	if err != nil {
-		return nil, err
-	}
-
-	if imageFilters.Contains("label") {
-		fltrs = append(fltrs, func(image images.Image) bool {
-			return imageFilters.MatchKVList("label", image.Labels)
-		})
+		return nil, nil, err
 	}
 
 	if imageFilters.Contains("dangling") {
 		danglingValue, err := imageFilters.GetBoolOrDefault("dangling", false)
 		if err != nil {
-			return nil, err
+			return nil, nil, err
 		}
 		fltrs = append(fltrs, func(image images.Image) bool {
 			return danglingValue == isDanglingImage(image)
 		})
 	}
 
-	return func(image images.Image) bool {
+	for _, fltrName := range []string{"label", "label!"} {
+		operator := "=="
+		if strings.HasSuffix(fltrName, "!") {
+			operator = "!="
+		}
+
+		for _, l := range imageFilters.Get(fltrName) {
+			k, v, found := strings.Cut(l, "=")
+			if !found {
+				return nil, nil, errdefs.InvalidParameter(fmt.Errorf("expected label=value"))
+			}
```

Comment:
Oh! @rumpl's comment made me start to look, and now I'm wondering if this is correct. The existing code uses `opts.Filters.MatchKVList()` (even wondering if we can use that here as well); https://github.com/moby/moby/blob/e8a79114b8c1782d5539421c9a6a1cd6fc5dfa65/daemon/images/image_list.go#L107-L116

In that logic, `--filter=label` (no value) means; filter based on the label being _present_ (regardless of its value), and if a value _is_ present for the filter (`--filter=label==value` or `--filter=label!=value`) only then it compares the value as well; https://github.com/moby/moby/blob/e8a79114b8c1782d5539421c9a6a1cd6fc5dfa65/api/types/filters/parse.go#L165-L173

Reply:
Yeah I already did fix it :P But didn't update the PR yet because these are completely different labels (see my previous comment).

Reply:
🙈 looks like I wasn't reading again ☺️ 😂 😬

Reply:
No worries, I should have updated the PR anyway 😄  Updated it now.